pkg/vault/contracts/authorizer/TimelockAuthorizerManagement.sol

```diff
@@ -201,6 +201,30 @@ abstract contract TimelockAuthorizerManagement is ITimelockAuthorizer {
         return _scheduledExecutions[scheduledExecutionId];
     }
 
+    function getNumScheduledExecutions() external view returns (uint256) {
+        return _scheduledExecutions.length;
+    }
+
+    function getScheduledExecutions(uint256 offset, uint256 limit, bool reverseOrder)
+        external
+        view
+        returns (ITimelockAuthorizer.ScheduledExecution[] memory items)
+    {
+        require(offset < _scheduledExecutions.length, "INVALID_OFFSET");
+      
+        uint256 rest = _scheduledExecutions.length - offset;
+        uint256 size = rest > limit ? limit : rest;
```

Comment:
```suggestion
        uint256 size = Math.min(rest, limit);
```

I find this makes reasoning about `size` easier (i.e. that `size` is never greater than `rest`, and that `limit` may reduce `rest`).